python/az/aro/azext_aro/_aad.py

```diff
@@ -51,6 +54,19 @@ def create_service_principal(self, app_id):
                 logger.warning("%s; retry %d of %d", ex, retries, max_retries)
                 time.sleep(10)
 
+    def delete_service_principal(self, app_id):
+        max_retries = 3
```

Comment:
Because we keep track of max_retries for a couple of functions, it would make more sense to make this a class-wide property.